audiomentations/augmentations/transforms.py

```diff
@@ -1151,7 +1152,200 @@ def apply(self, samples, sample_rate):
+    ):
+        """
+        :params selected_samples: (batch_size, num_channels, num_samples)
+        """
+        batch_size, _, num_samples = selected_samples.shape
+
+        self.parameters["cutoff_freq"] = np.random.uniform(
+            low=self.min_cutoff_freq,
+            high=self.max_cutoff_freq,
+            size=batch_size
+        )
+
+    def apply_transform(self, selected_samples: np.array, sample_rate: int = None):
+        batch_size, num_channels, num_samples = selected_samples.shape
+
+        if sample_rate is None:
+            sample_rate = self.sample_rate
+
+        cutoffs_as_fraction_of_sample_rate = (
+            self.parameters["cutoff_freq"] / sample_rate
+        )
+        # TODO: Instead of using a for loop, perform batched compute to speed things up
+        for i in range(batch_size):
+            selected_samples[i] = low_pass_filter(
+                selected_samples[i], cutoffs_as_fraction_of_sample_rate[i].item()
+            )
+
+        return selected_samples
+
+
+class HighPassFilter(BaseWaveformTransform):
+    """
+    Apply high-pass filtering to the input audio.
+    """
+
+    supports_multichannel = True
+    requires_sample_rate = True
+
+    def __init__(
+        self,
+        min_cutoff_freq=20,
+        max_cutoff_freq=2400,
+        p: float = 0.5
+    ):
+        """
+        :param min_cutoff_freq: Minimum cutoff frequency in hertz
+        :param max_cutoff_freq: Maximum cutoff frequency in hertz
+        :param p:
+        """
+        super().__init__(p)
+
+        self.min_cutoff_freq = min_cutoff_freq
+        self.max_cutoff_freq = max_cutoff_freq
+        if self.min_cutoff_freq > self.max_cutoff_freq:
+            raise ValueError("min_cutoff_freq must not be greater than max_cutoff_freq")
+
+    def randomize_parameters(
+        self, selected_samples: np.array, sample_rate: int = None
+    ):
+        """
+        :params selected_samples: (batch_size, num_channels, num_samples)
+        """
+        batch_size, _, num_samples = selected_samples.shape
+
+        self.parameters["cutoff_freq"] = np.random.uniform(
+            low=self.min_cutoff_freq,
+            high=self.max_cutoff_freq,
+            size=batch_size
+        )
+
+    def apply_transform(self, selected_samples: np.array, sample_rate: int = None):
+        batch_size, num_channels, num_samples = selected_samples.shape
+
+        if sample_rate is None:
+            sample_rate = self.sample_rate
+
+        cutoffs_as_fraction_of_sample_rate = (
+            self.parameters["cutoff_freq"] / sample_rate
+        )
+        # TODO: Instead of using a for loop, perform batched compute to speed things up
+        for i in range(batch_size):
+            selected_samples[i] = high_pass_filter(
+                selected_samples[i], cutoffs_as_fraction_of_sample_rate[i].item()
+            )
+
+        return selected_samples   
+
+
+class BandPassFilter(BaseWaveformTransform):
```

Comment:
I would feel more comfortable having two parameters center frequency and bandwidth instead of two parameters HPF cutoff and LPF cutoff. That also eliminates any potential issues where low cutoff > high cutoff, leading to a silent output.

It probably makes sense to have the bandwidth be related to the chosen center frequency. Maybe the bandwidth can be a float that denotes the "ratio of the center frequency"?

Some bandpass filter developers like to have a bandwidth parameter with inverse behavior, and they call it Q. An example of that can be found here: https://github.com/facebookresearch/AugLy/blob/main/augly/audio/transforms.py#L561

Reply:
Which parameters should be selected randomly in that case? Center_freq, Q, or both of them?

Reply:
Center freq and Q could both be selected randomly

Reply:
People who want e.g. a constant Q can set min Q and max Q to the same value :)

Reply:
OK, got it.

Reply:
Done!